mathesar/api/display_options.py

```diff
@@ -0,0 +1,18 @@
+from mathesar.database.types import MathesarTypeIdentifier
+
+DISPLAY_OPTIONS_BY_TYPE_IDENTIFIER = {
+    MathesarTypeIdentifier.BOOLEAN.value:
+        {
+            "options": [{"name": "input", "type": "string",
+                         "enum": ['dropdown', 'checkbox']},
+                        {'name': "custom_labels", "type": "object",
+                         "items": [{"name": "TRUE", "type": "string"},
+                                   {'name': "FALSE", "type": "string"}]}]
+
+        },
+    MathesarTypeIdentifier.NUMBER.value:
+        {
+            "options": [{"name": "show_as_percentage", "type": "boolean"},
+                        {"name": "locale", "type": "string"}]
```

Comment:
Should the backend be aware of localization preferences? My instinct is to have the frontend take care of that. So you could have two people working on the same table using different locales.

Reply:
Display options are used to store (default) representational preferences that will be used by the frontend, it won't be used by the backend. Please have a look at @kgodey [comment](https://github.com/centerofci/mathesar/issues/392#issuecomment-969168596) that concludes this

Reply:
Interesting, I wonder in what scenario a frontend would find backend-defined defaults useful.

Reply:
Display options are not defined/generated by the backend, it is set by the user on the frontend using the [`/columns/` API](https://github.com/centerofci/mathesar/blob/f4c2b32d3fd8320f8880a54c951243ae4254f969/mathesar/api/serializers/columns.py#L42). The display options returned as part of `/types/` API is used by the client to know about the available display options for that particular type

Reply:
I had misunderstood what I was reading. Thanks for bearing with me!